### PR TITLE
fix(wadouri): custom loaders should register their schema, and add new debounce listener to event target

### PR DESCRIPTION
**Pull request by sedghi**
### Context

We are reverting changes here https://github.com/cornerstonejs/cornerstone3D/pull/1154/files since it is causing troubles. 

Since it is wrong after further consideration with Bill Wallace, for your custom loader, you should register the scheme with it and use your custom scheme in your URLs.

CC: @jlopes90 

#### PR

- [] My Pull Request title is descriptive, accurate and follows the
  semantic-release format and guidelines.

#### Code

- [] My code has been well-documented (function documentation, inline comments,
  etc.)

- [] I have run the `yarn build:update-api` to update the API documentation, and have
  committed the changes to this PR. (Read more here https://www.cornerstonejs.org/docs/contribute/update-api)

#### Public Documentation Updates

- [] The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [] "OS: 
- [] "Node version: 
- [] "Browser:
  

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch

### CODE REVIEW

File: packages/core/src/eventTarget.ts [hunk 1]
```diff
@@ -39,6 +42,49 @@ class CornerstoneEventTarget implements EventTarget {
     this.listeners[type].push(callback);
   }
 
+  public addEventListenerDebounced(type, callback, delay) {
```
Comment (wayfarer3130) [inline packages/core/src/eventTarget.ts:45]: Docs please - would like to know something about how it is debounced

Comment (wayfarer3130) [reply packages/core/src/eventTarget.ts:45]: Would be nice to see this used in at least one example please.

**Review by wayfarer3130 — APPROVED** [review]

**Comment by shunia** [comment]
> ... for your custom loader, you should register the scheme with it and use your custom scheme in your URLs.

Please, share some detailed docs about how to implement a complete example on how to create a custom loader with just pure **ArrayBuffer**. 

It might be right to remove the wrong use cases to `wadouri.loadImage` with a custom loader since it's not designed for it, but just as [this comment](https://github.com/cornerstonejs/cornerstone3D/pull/990#issuecomment-1892221409) described, this is the only way I found that could easily use cornerstone with my dicom data from some custom sources like from a **Websocket**.

I don't want to say it's blocking ppl from using a custom loader but the reality here is that there's no real use cases or examples for us to learn from at all. 

It feels really helpless.

**Comment by sedghi** [comment]
@shunia please see our docs

https://www.cornerstonejs.org/docs/how-to-guides/custom-image-loader

**Comment by shunia** [comment]
> @shunia please see our docs
> 
> https://www.cornerstonejs.org/docs/how-to-guides/custom-image-loader

How to create the image object? Is there any utility function provided for convenience?

**Comment by sgielen** [comment]
> @shunia please see our docs
> 
> https://www.cornerstonejs.org/docs/how-to-guides/custom-image-loader

Please bear with me if I'm wrong here -- but the docs, unfortunately, leave out the difficult part: they simply use `createImageObject` which isn't defined anywhere ("logic omitted" has the same energy as "I can prove this, but the margin of my scientific paper is too small for it"), and they don't mention metadata at all. Those are the two things we used WADOImageLoader for, which is now impossible. Could you please advise?

**Pull request merged**